datajunction-server/datajunction_server/api/dimensions.py

```diff
@@ -18,7 +19,7 @@
 from datajunction_server.utils import get_session
 
 _logger = logging.getLogger(__name__)
-router = APIRouter(tags=["dimensions"])
+router = APIRouter(tags=["dimensions"], dependencies=[Depends(HTTPBearer())])
```

Comment:
Should this be `dependencies=[Depends(HTTPBearer())] if settings.secret else [],`?

Reply:
Fixed by your other suggestion to move to a central `SecureAPIRouter`